wiremock/deployment/src/main/java/io/quarkiverse/openapi/wiremock/generator/deployment/wrapper/OpenApiWiremockGeneratorWrapper.java

```diff
@@ -0,0 +1,91 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkiverse.openapi.wiremock.generator.deployment.wiremock.OpenApi2WiremockMapper;
+import io.quarkiverse.openapi.wiremock.generator.deployment.wiremock.model.Stubbing;
+import io.quarkus.bootstrap.prebuild.CodeGenException;
+import io.smallrye.config.common.utils.StringUtil;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+
+public class OpenApiWiremockGeneratorWrapper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenApiWiremockGeneratorWrapper.class);
+    public static final String MAPPINGS_DIR = "mappings";
+    private static final String MAPPINGS = "mappings";
+    private static final String WIREMOCK_OUTPUT_FILE = "wiremock-stubbings.json";
+    private static final OpenAPIV3Parser OPENAPI_PARSER_INSTANCE = new OpenAPIV3Parser();
+    public static final ObjectMapper OBJECT_MAPPER_INSTANCE = new ObjectMapper();
+    private final Path outputDir;
+    private final Path inputDir;
+
+    public OpenApiWiremockGeneratorWrapper(final Path inputDir, final Path outputDir) {
+        this.inputDir = inputDir;
+        this.outputDir = outputDir;
+    }
+
+    public void generate() throws CodeGenException, IOException {
+        if (Files.isDirectory(inputDir) && !PathUtils.isEmpty(this.inputDir)) {
+            try (Stream<Path> stream = Files.walk(this.inputDir)) {
+                Path mappingsDir = createOutputMappingsDir(this.outputDir);
+                stream.filter(Files::isRegularFile)
+                        .forEach(generateWiremockStubbingFile(mappingsDir));
+            } catch (IOException e) {
+                throw new CodeGenException(
+                        "Failed to generate Wiremock stubbing from OpenApi files in " + this.inputDir.toAbsolutePath(), e);
+            }
+        }
+    }
+
+    private Path createOutputMappingsDir(Path outDir) throws IOException {
+        Path mappingsDir = resolveOutDir(outDir);
+        if (!Files.exists(mappingsDir)) {
+            Files.createDirectory(mappingsDir);
+        }
+        return mappingsDir;
+    }
+
+    private Consumer<Path> generateWiremockStubbingFile(Path mappingsDir) {
+        return file -> {
+            OpenAPI openAPI = OPENAPI_PARSER_INSTANCE.read(file.toString());
+            OpenApi2WiremockMapper openApi2WiremockMapper = new OpenApi2WiremockMapper(openAPI);
+            List<Stubbing> stubbings = openApi2WiremockMapper.generateWiremockStubbings();
+
+            try {
+
+                byte[] json = OBJECT_MAPPER_INSTANCE.writeValueAsBytes(Map.of(MAPPINGS, stubbings));
+
+                Path wiremockStubbings = Files
+                        .createFile(mappingsDir.resolve(generateFinalFilename(file)));
+
+                Files.write(wiremockStubbings, json);
+
+            } catch (IOException e) {
+                LOGGER.warn("Failed to generate Wiremock stubs for file " + file.toAbsolutePath(), e);
```

Comment:
Are you sure we don't need to abort the entire generation here?

Reply:
We are generating one Wiremock stub for one OpenAPI specification file, the generation is independent, if one fails does not affect the another one, in your experience, is better make the generation dependent?

Reply:
In general, I prefer failing "atomically".
In this particular case, it doesn't seem like users would need to generate just a few stubs. I think they'd like to generate them all. (I would, as a user. But I may be wrong.)
Also, if we just log the exception, users may not notice it occurred.
I don't have a strong opinion here, but I would generate them all or fail (not just log the error).

Reply:
It help me to think, makes sense!